src/Databases/IDatabase.h

```diff
@@ -275,6 +284,10 @@ class IDatabase : public std::enable_shared_from_this<IDatabase>
         return nullptr;
     }
 
+    /// Daisy: starts.
+    std::tuple<String, String> astToQueryAndEngineStringImpl(ASTPtr ast, ContextPtr context) const;
```

Comment:
i prefer to rename it as `std::pair<String, String> parseTableCreateQueryAndEngineFullStringFrom(ASTPtr ast, ...)`